### DISCUSSION TITLE
Improvements

### DISCUSSION THREAD

**Discussion by stevoh6** [flobz/psa_car_controller#469] [Ideas]
Hi, please comment what do you want first to implement :)
I have couple of ideas, since Im programmer and Im using this awesome software, I want to contribute.

- fix update charging price
- trips price calculation of each trip (on the fly, they will be only displayed,not stored)
- charging limiter Ac vs Dc
- charging limiter inprovements - based on events not on time -i need to chceck what can be improved and how
- mqtt integration with full home assistant discovery
- live consumption data for ABRP (every time psa controller get new position, it calculate/update current  trip consumption and send it to ABRP) for new trips it use avg cunsumption from all routes, also this needs to be possible to configure.

**Comment by marc6901** [comment]
Hi, this improvement sounds interesting:

1. mqtt integration with full home assistant discovery
